source/Request.cpp

```diff
@@ -66,9 +66,8 @@ namespace HTTP{
 
     bool Request::is_authority_uri(const std::string& uri)
     {
-        if(true)
-        {
-            return (m_uri.get_path() == empty_string);
+        if(std::regex_match( uri.begin(), uri.end(), std::regex(R"([a-zA-Z0-9+:\_\.@]*)"))){ 
```

Comment:
I've wrote this function in this way because our Uri parser is not smart enough to parse correctly string witch contains only authority.  But problem with this is that we can't save that authority from request line. The same case occurs when we have asterisk as request_uri.